modules/post/windows/gather/forensics/fanny_bmp_check.rb

```diff
@@ -0,0 +1,65 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/loneicewolf/metasploit_fanny_check_module
+##
+
+require 'msf/core'
+require 'msf/core/post/common'
+require 'msf/core/post/windows/priv'
+class MetasploitModule < Msf::Post
+
+        include Msf::Post::Common
+        include Msf::Post::Windows::Registry
+
+
+        def initialize(info={})
+                super( update_info( info,
+                                 'Name' => 'FannyBMP Registry Check',
+                                 'Description' => %q{This module searches for the Fanny.bmp worm related registry keys},
+                                 'License' => MSF_LICENSE,
+                                 'Author' => [ 'William M.'],
+                                 'Platform' => [ 'win' ],
+                                 'SessionTypes' => [ 'meterpreter' ],
```

Comment:
So I don't know if the Fanny.bmp worm affects systems newer than XP, but FWIW we recently had the discussion of dropping meterpreter support for Windows XP (see #14473). As it is now we only support XP SP2 and newer but that could easily change in the next year or two. The good news is that it looks like your module should support shell sessions.

You should update the SessionTypes to include shell and test this with shell sessions because it's very possible that in a year that will be the only session type support by post modules for Windows XP.

Reply:
@smcintyre-r7 Roger that. Thanks for dropping by.
Noted down.
//Will.

Reply:
Yeah, update that line to `'SessionTypes' => [ 'meterpreter', 'shell' ]` which will make it compatible with shell sessions like `windows/shell/reverse_tcp`. Then get a session using that payload which will be a shell session, background it using `Ctrl+Z`, and run this module using that session to make sure it still works. Nothing caught my eye that I think would cause it to break since you're just enumerating the registry keys, but I could be wrong.

Reply:
> Nothing caught my eye that I think would cause it to break since you're just enumerating the registry keys, but I could be wrong.

I *think* `sysinfo['Computer']` is Meterpreter only, but I could be wrong.

Reply:
Trying what you said too,  I probably just made a quick-error but ill check

Reply:
Additional comment: I could just used a new (1 only) session,  but as you see I used  **Session 4** (4 shell x86/windows ...)

Reply:
Sorry everyone, I totally forgot to **make the change** at the file (With the  meterpreter ++ shell session)  
, (I just changed my own, local file) and showed the results,  (How stupid can one be?) sorry. Will do now, and  also begin with the  Docs and the pointed out  Improvements in the module (Array, etc.)